docs/aws/audit/redshiftmonitoring/rules/not_using_default_port.mdx

```diff
@@ -24,6 +24,231 @@ PCIDSS
+
+2. List all Redshift clusters: Use the following AWS CLI command to list all the Redshift clusters in your account:
+
+   ```
+   aws redshift describe-clusters --query 'Clusters[*].[ClusterIdentifier,Endpoint.Port]' --output text
+   ```
+
+   This command will return a list of all Redshift clusters along with their ports. 
+
+3. Check the port of each cluster: The default port for Redshift clusters is 5439. If the port returned by the above command for any cluster is 5439, it means that the cluster is using the default port.
+
+4. Automate the process: If you have a large number of clusters, it might be more efficient to automate the process using a Python script. You can use the `boto3` library to interact with AWS services. Here is a simple script that checks if any Redshift cluster is using the default port:
+
+   ```python
+   import boto3
+
+   client = boto3.client('redshift')
+
+   response = client.describe_clusters()
+
+   for cluster in response['Clusters']:
+       if cluster['Endpoint']['Port'] == 5439:
+           print(f"Cluster {cluster['ClusterIdentifier']} is using the default port")
+   ```
+   
+   This script will print the identifiers of all clusters that are using the default port.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3) on your local system. Boto3 allows you to directly create, update, and delete AWS services from your Python scripts.
+
+```python
+pip install boto3
+```
+
+2. Import the necessary modules and create a session using your AWS credentials.
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    region_name='YOUR_REGION'
+)
+```
+
+3. Create a Redshift client using the session.
+
+```python
+redshift = session.client('redshift')
+```
+
+4. Use the `describe_clusters` method to retrieve information about all Redshift clusters. Then, iterate over each cluster and check the `Endpoint` attribute to see if the `Port` is set to the default value (5439). If it is, print a message indicating that the cluster is using the default port.
+
+```python
+clusters = redshift.describe_clusters()
+
+for cluster in clusters['Clusters']:
+    if cluster['Endpoint']['Port'] == 5439:
+        print(f"Cluster {cluster['ClusterIdentifier']} is using the default port.")
+```
+
+This script will help you identify any Redshift clusters that are using the default port. However, it's important to note that using the default port isn't necessarily a misconfiguration - it depends on your specific use case and security requirements.
+</Accordion>
+
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>